crates/bevy_ecs/src/component.rs

```diff
@@ -1135,73 +1138,538 @@ impl ComponentCloneBehavior {
     }
 }
 
-/// Stores metadata associated with each kind of [`Component`] in a given [`World`].
+/// Coordinates a registration that is reserved, but not registered.
+trait QueuedComponentRegistration {
```

Comment:
Why does this trait exist? It seems like its only purpose is to call `register` once. Could we not just replace it with `Box<dyn FnOnce(&mut ComponentsRegistrator, ComponentId)>`?